diesel_cli/src/infer_schema_internals/inference.rs

```diff
@@ -14,7 +14,9 @@ static RESERVED_NAMES: &[&str] = &[
     "type", "typeof", "unsafe", "unsized", "use", "virtual", "where", "while", "yield",
 ];
 
-fn is_reserved(name: &str) -> bool {
+static UNMAPPABLE_CHARS: &[char] = &['-', ' '];
```

Comment:
SQL table names can use other characters that would not translate well to a Rust identifier (e.g. a colon `:`).

Rust identifiers are restricted to `[a-z A-Z 0-9 _]`, so I think that a more robust approach would be to map any character outside of this range to an underscore.